packages/shell-api/src/collection.ts

```diff
@@ -159,26 +160,27 @@ export default class Collection extends ShellApiWithMongoClass {
    */
   async aggregate(
     pipeline: Document[],
-    options: Document & { explain?: never }
-  ): Promise<AggregationCursor>;
+    options: AggregateOptions & { explain: ExplainVerbosityLike }
```

Comment:
I realize removing `& {explain?: never}` seems not intuitive but this actually was causing issues in cases where.

```ts
const options: AggregateOptions = {};
// This would complain that neither of the overrides match the type of options
db.coll.aggregate({}, options);
```
With this setup, what we're looking for still happens as the first overload takes priority so when explain is defined, it'd pick this and return `Document` as expected

Reply:
Hm, I tried again with the never and now I am not seeing it; a bit weird

Reply:
This exact example seems to work for me – can you share a reproduction/the actual error for this case?

Reply:
@addaleax actually I only see it in the database.ts file; adding `& {explain: never}` leads to the sql function complaining about it, but maybe there's something specific there

Reply:
```
  Overload 1 of 3, '(pipeline: Document[], options: AggregateOptions & { explain: ExplainVerbosityLike; }): Promise<Document>', gave the following error.
    Argument of type 'AggregateOptions | undefined' is not assignable to parameter of type 'AggregateOptions & { explain: ExplainVerbosityLike; }'.
      Type 'undefined' is not assignable to type 'AggregateOptions & { explain: ExplainVerbosityLike; }'.
        Type 'undefined' is not assignable to type 'AggregateOptions'.
  Overload 2 of 3, '(pipeline: Document[], options: (AggregateOptions & { explain: never; }) | undefined): Promise<AggregationCursor>', gave the following error.
    Argument of type 'AggregateOptions | undefined' is not assignable to parameter of type '(AggregateOptions & { explain: never; }) | undefined
```
I think it's specifically the case where `AggregateOptions | undefined`?

Reply:
Yeah what I was describing is specifically a problem with
```
    await this.aggregate([], {} as AggregateOptions | undefined);
```

Reply:
Fwiw, if you "just" want to fix this here, you could probably make that work by using `this.aggregate([...], options ?? {})` in the .sql() function, but I don't know if I have a strong preference for that really